docs/src/tutorials/device-monitoring.md

```diff
@@ -0,0 +1,154 @@
+You will have to install and configure [`collectd`](https://collectd.org/) first.
+
+To install collectd, follow the [collectd installation process](https://collectd.org/download.shtml)
+that is specific to your device. On a Debian or Ubuntu linux:
+
+``` shell
+sudo apt-get install collectd-core
+```
+
+## Configure `collectd`
+
+### TLDR; Just want it running
+
+Thin-edge.io provides a [basic `collectd` configuration](https://github.com/thin-edge/thin-edge.io/blob/main/configuration/contrib/collectd/tedge-collectd.conf)
+that can be used to collect cpu, memory and disk metrics.
+
+Simply copy that file to the main collectd configuration file and restart the daemon
+(it might be good to keep a copy of the original configuration).
+
+``` shell
+sudo cp /etc/tedge/contrib/collectd/collectd.conf sudo cp /etc/tedge/contrib/collectd/collectd.conf.backup
+sudo cp /etc/tedge/contrib/collectd/collectd.conf /etc/collectd/collectd.conf
+sudo systemctl restart collectd
+```
+
+### `Collectd.conf`
+
+The behavior of collectd is defined by the 
+[`collectd.conf` configuration file](https://collectd.org/documentation/manpages/collectd.conf.5.shtml)
+(usually located at `/etc/collectd/collectd.conf`)
+
+__Important notes__ You can enable or disable the collectd plugins of your choice, but with some notable exceptions:
+1. __MQTT must be enabled__.
+   * Thin-edge.io expects the collectd metrics to be published on the local MQTT bus.
+     Hence, you must enable the [MQTT write plugin of collectd](https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_mqtt).
+   * The MQTT plugin is available on most distribution of `collectd`, but this is the case on MacOS using homebrew.
+     If you are missing the MQTT plugin, please recompile `collectd` to include the MQTT plugin.
+     See [https://github.com/collectd/collectd](https://github.com/collectd/collectd) for details.
+   * Here is a config snippet to configure the MQTT write plugin:
+     ```
+        LoadPlugin mqtt
+        
+        <Plugin mqtt>
+            <Publish "tedge">
+                Host "localhost"
+                Port 1883
+                ClientId "tedge-dm"
+            </Publish>
+        </Plugin>
+     ```
+2. __RRDTool and CSV might be disabled__
+   * The risk with these plugins is to run out of disk space on a small device.
+   * With thin-edge.io the metrics collected by `collectd` are forwarded to the cloud,
+     hence it makes sense to [disable Local storage](https://github.com/collectd/collectd/issues/2668).
+   * For that, simply comment out these two plugins:
+    ```
+       #LoadPlugin rrdtool
+       #LoadPlugin csv
+    ```
+3. __Cherry-pick the collected metrics__
+   * `Collectd` can collect a lot of detailed metrics,
+      and it doesn't always make sense to forward all these data to the cloud.
+   * Here is a config snippet that uses the `match_regex` plugin to select the metrics of interest:
+    ```
```

Comment:
Do we need to describe what this configuration does? As in "filters out every metric emitted by the `memory` plugin other than the `used` metric"

Reply:
Added